internal/datastore/spanner/reader.go

```diff
@@ -149,12 +149,12 @@ func (sr spannerReader) ReadNamespaceByName(ctx context.Context, nsName string)
 	return ns, revisionFromTimestamp(updated), nil
 }
 
-func (sr spannerReader) ListAllNamespaces(ctx context.Context) ([]*core.NamespaceDefinition, error) {
+func (sr spannerReader) ListAllNamespaces(ctx context.Context) ([]datastore.RevisionedNamespace, error) {
 	iter := sr.txSource().Read(
 		ctx,
 		tableNamespace,
 		spanner.AllKeys(),
-		[]string{colNamespaceConfig},
+		[]string{colNamespaceConfig, colNamespaceTS},
```

Comment:
same, do we need to consider an index here, or is this metadata available always for all fetched rows in Spanner?

Reply:
It should be available for all of them, but I'm not 100% certain about Spanner. @jakedt any thoughts?